node/network/src/protocol.rs

```diff
@@ -133,10 +162,21 @@ impl Protocol {
                     warn!(target: "network", "Requesting snapshot from peer {} who is not an authority. {:?}", peer_id, auth_map);
                 }
             }
-            Message::PayloadResponse(_request_id, missing_payload) => {
-                // TODO: check request id and pull block_index from there.
+            Message::PayloadResponse(request_id, missing_payload) => {
+                match self.requests.read().expect(POISONED_LOCK_ERR).get(&request_id) {
```

Comment:
```if Some(RequestType::PayloadRequest(hash)) = self.requests.read().expect(POISONED_LOCK_ERR).pop(&request_id) {
} else {
self.peer_manager.ban_peer(&peer_id)
}```

clippy should be recommending this, no?

Reply:
In the case when `else` is needed, I think there's no difference between the two.